inference-engine/src/transformations/src/transformations/control_flow/unroll_tensor_iterator.cpp

```diff
@@ -19,13 +19,19 @@ NGRAPH_RTTI_DEFINITION(ngraph::pass::UnrollTensorIterator, "UnrollTensorIterator
 bool ngraph::pass::UnrollTensorIterator::run_on_function(std::shared_ptr<ngraph::Function> f) {
     RUN_ON_FUNCTION_SCOPE(UnrollTensorIterator);
     for (const auto& op : f->get_ops()) {
-        auto ti = std::dynamic_pointer_cast<ngraph::opset4::TensorIterator>(op);
-        if (!ti || transformation_callback(ti)) {
+        auto sub_graph_op = std::dynamic_pointer_cast<ngraph::op::util::SubGraphOp>(op);
+        if (!sub_graph_op || transformation_callback(sub_graph_op)) {
             continue;
         }
 
-        const auto& function = ti->get_body();
-        const auto num_iter = ti->get_num_iterations();
+        const auto &function = sub_graph_op->get_function();
+        int64_t num_iter = -1;
+        const auto &loop = std::dynamic_pointer_cast<ngraph::opset6::Loop>(sub_graph_op);
+        if (loop) {
+            num_iter = loop->get_num_iterations();
```

Comment:
Should we put get_num_iterations to the SubGraphOp interface?

Reply:
done